src/Conversion/ONNXToKrnl/ML/CategoryMapper.cpp

```diff
@@ -231,6 +231,29 @@ struct ONNXCategoryMapperOpLowering : public ConversionPattern {
     return res;
   }
 
+  Value loadElement(Value memref, ValueRange loopInd, Type elementType,
+      int64_t rank, const KrnlBuilder &createKrnl) const {
+    Value inputElem;
+    TypeSwitch<Type>(elementType)
+        .Case<IntegerType>(
+            [&](IntegerType) { inputElem = createKrnl.load(memref, loopInd); })
+        .Case<StringType>([&](StringType stringType) {
+          OpBuilder &builder = createKrnl.getBuilder();
+          Location loc = createKrnl.getLoc();
+          auto zero = builder.create<arith::ConstantOp>(
```

Comment:
We have a MathBuilder `  Value constant(Type type, double val) const;` that is a bit more compact

Reply:
done